keras/layers/convolutional_recurrent.py

```diff
@@ -334,7 +334,8 @@ def build(self, input_shape):
         if isinstance(input_shape, list):
             input_shape = input_shape[0]
         batch_size = input_shape[0] if self.stateful else None
-        self.input_spec[0] = InputSpec(shape=(batch_size,) + input_shape[1:])
+        input_length = input_shape[1] if self.unroll else None
```

Comment:
Mistake in my previous statement: unrolling should not actually affect the input spec. The input length should always be None, because even an unrolled layer can be applied to inputs of different lengths (as long as all lengths are static). The correct behavior here is to set `InputSpec(shape=(batch_size, None) + input_shape[2:])`